cli_tools/gce_ovf_export/exporter/exporter.go

```diff
@@ -0,0 +1,210 @@
+	storageClient, err := storageutils.NewStorageClient(ctx, logger)
+	if err != nil {
+		return nil, err
+	}
+	computeClient, err := createComputeClient(&ctx, params)
+	if err != nil {
+		return nil, err
+	}
+	metadataGCE := &computeutils.MetadataGCE{}
+
+	paramValidator := NewOvfExportParamValidator(computeClient)
+	paramPopulator := NewPopulator(metadataGCE, storageClient,
+		storageutils.NewResourceLocationRetriever(metadataGCE, computeClient),
+		storageutils.NewScratchBucketCreator(ctx, storageClient),
+	)
+	if err := ValidateAndPopulateParams(params, paramValidator, paramPopulator); err != nil {
+		return nil, err
+	}
+	inspector, err := commondisk.NewInspector(params.DaisyAttrs(), params.Network, params.Subnet)
+	if err != nil {
+		return nil, daisy.Errf("Error creating disk inspector: %v", err)
+	}
+	return &OVFExporter{
+		storageClient:          storageClient,
+		computeClient:          computeClient,
+		mgce:                   metadataGCE,
+		bucketIteratorCreator:  &storageutils.BucketIteratorCreator{},
+		Logger:                 logger,
+		params:                 params,
+		loggableBuilder:        service.NewOvfExportLoggableBuilder(),
+		ovfDescriptorGenerator: NewOvfDescriptorGenerator(computeClient, storageClient, *params.Project, params.Zone),
+		manifestFileGenerator:  NewManifestFileGenerator(storageClient),
+		inspector:              inspector,
+		instanceDisksExporter:  NewInstanceDisksExporter(computeClient, storageClient),
+		instanceExportPreparer: NewInstanceExportPreparer(),
+		instanceExportCleaner:  NewInstanceExportCleaner(),
+	}, nil
+}
+
+// ValidateAndPopulateParams validate and populate OVF export params
+func ValidateAndPopulateParams(params *ovfexportdomain.OVFExportParams,
+	paramValidator ovfexportdomain.OvfExportParamValidator,
+	paramPopulator ovfexportdomain.OvfExportParamPopulator) error {
+	if err := paramValidator.ValidateAndParseParams(params); err != nil {
+		return err
+	}
+	if err := paramPopulator.Populate(params); err != nil {
+		return err
+	}
+	return nil
+}
+
+// creates a new Daisy Compute client
+//TODO: consolidate with ovf_importer.createComputeClient
+func createComputeClient(ctx *context.Context, params *ovfexportdomain.OVFExportParams) (daisycompute.Client, error) {
+	computeOptions := []option.ClientOption{option.WithCredentialsFile(params.Oauth)}
+	if params.Ce != "" {
+		computeOptions = append(computeOptions, option.WithEndpoint(params.Ce))
+	}
+
+	computeClient, err := daisycompute.NewClient(*ctx, computeOptions...)
+	if err != nil {
+		return nil, err
+	}
+	return computeClient, nil
+}
+
+func (oe *OVFExporter) buildLoggable() service.Loggable {
+
+	exportedDisksSourceSizes := make([]int64, len(oe.exportedDisks))
+	exportedDisksTargetSizes := make([]int64, len(oe.exportedDisks))
+	for i, exportedDisk := range oe.exportedDisks {
+		exportedDisksSourceSizes[i] = exportedDisk.Disk.SizeGb
+		exportedDisksTargetSizes[i] = exportedDisk.GcsFileAttrs.Size / bytesPerGB
```

Comment:
shouldn't it be (exportedDisk.GcsFileAttrs.Size-1) / bytesPerGB + 1?
Though it's not a problem when PD never assign a non-integer GB size.

Reply:
Done